hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/EncryptedFileSystem.java

```diff
@@ -88,6 +91,19 @@ public FSDataOutputStream create(Path path, FsPermission permission,
         FSDataOutputStream encryptedStream =
                 fs.create(path, permission, overwrite, bufferSize, replication, blockSize, progress);
 
+        return storeKey(encryptedStream, path);
+    }
+
+    @Override
+    public FSDataOutputStream create(Path path, FsPermission permission, EnumSet<CreateFlag> flags, int bufferSize,
+            short replication, long blockSize, Progressable progress, ChecksumOpt checksumOpt) throws IOException {
+        FSDataOutputStream encryptedStream =
+                fs.create(path, permission, flags, bufferSize, replication, blockSize, progress, checksumOpt);
+
+        return storeKey(encryptedStream, path);
+    }
+
+    private FSDataOutputStream storeKey(FSDataOutputStream encryptedStream, Path path) throws IOException {
```

Comment:
`storeKey` -> `encrypt`

Reply:
Done, though I think it's a bit confusing that now we're saying `return encrypt(encryptedStream, path);` in the create methods. I think the `encryptedStream` variable might not be named correctly since it's just the result of calling `fs.create`, i.e., the delegate won't actually do the encryption, right?

Reply:
yeah should rename the variable as well. `outputStream` is fine.